src/nvim/aucmd.c

```diff
@@ -50,12 +52,46 @@ void aucmd_schedule_focusgained(bool gained)
 static void do_autocmd_focusgained(bool gained)
 {
   static bool recursive = false;
+  static Timestamp last_time = (time_t)0;
+  bool need_redraw = false;
 
   if (recursive) {
     return;  // disallow recursion
   }
   recursive = true;
-  apply_autocmds((gained ? EVENT_FOCUSGAINED : EVENT_FOCUSLOST),
-                 NULL, NULL, false, curbuf);
+  need_redraw |= apply_autocmds((gained ? EVENT_FOCUSGAINED : EVENT_FOCUSLOST),
+                                NULL, NULL, false, curbuf);
+
+  // When activated: Check if any file was modified outside of Vim.
+  // Only do this when not done within the last two seconds (could get
+  // several events in a row).
+
+  if (gained && last_time + (Timestamp)2 < os_time()) {
```

Comment:
we can keep the time check. because the filesystem timestamps seem to be in seconds, it makes no sense to check between less than a second but I think we could use os_now for moree precision and maybe check around 1500ms.

Reply:
Sure thing!